Microsoft.HealthVault/Thing/ThingKey.cs

```diff
@@ -106,7 +109,17 @@ public ThingKey(Guid id)
         /// is always equal to the <see cref="Id"/> of that item.
         /// </remarks>
         ///
-        public Guid VersionStamp => _versionStamp;
+        public Guid VersionStamp
+        {
+            get { return _versionStamp; }
+            set
+            {
+                if(value != Guid.Empty)
```

Comment:
I think we should just not assign it. There is no reason to fail at this point. 

---
In reply to: [121810435](https://github.com/Microsoft/healthvault-dotnetstandard-sdk/pull/57#discussion_r121810435) [](ancestors = 121810435)